### PR TITLE
Add counter-vanilla example

### PR DESCRIPTION
**Pull request by gaearon**
The new Counter Vanilla example is aimed to dispel the myth that Redux requires Webpack, React, hot reloading, sagas, action creators, constants, Babel, npm, CSS modules, decorators, fluent Latin, an Egghead subscription, a PhD, or an Exceeds Expectations O.W.L. level.

Nope, it's just HTML, some artisanal `<script>` tags, and plain old DOM manipulation. Enjoy!

### CODE REVIEW

File: examples/counter-vanilla/index.html [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Redux basic example</title>
+    <script src="https://npmcdn.com/redux@latest/dist/redux.min.js"></script>
+  </head>
+  <body>
+    <div>
+      <p>
```
Comment (gaearon) [inline examples/counter-vanilla/index.html]: I haven't written HTML in ages. It looks so much like JSX! Oh wait

File: examples/counter-vanilla/index.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      var valueEl = document.getElementById('value')
+
+      function render() {
+        valueEl.innerHTML = store.getState().toString()
+      }
+
+      render()
+      store.subscribe(render)
+
+      document.getElementById('increment')
```
Comment (danmartinez101) [inline examples/counter-vanilla/index.html]: Reminds me of a CycleJS app <3

File: examples/counter-vanilla/index.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Redux basic example</title>
+    <script src="https://npmcdn.com/redux@latest/dist/redux.min.js"></script>
```
Comment (gaearon) [inline examples/counter-vanilla/index.html]: I almost had a JS fatigue here but thank goodness @mjackson made npmcdn

Comment (danmartinez101) [inline examples/counter-vanilla/index.html]: Oh but you aren't using bower and webpack here, how can this possibly work?

**Comment by imkost** [comment]
I wonder how `render` function would look like for some bigger Vanilla application. Imagine, there is a Vanilla application with 5 fields in state object and with 5 ui widgets to represent them. So render function would look like:

```js
function render() {
  var state = store.getState();

  rerenderWidget1(state.widget1);
  rerenderWidget2(state.widget2);
  rerenderWidget3(state.widget3);
  rerenderWidget4(state.widget4);
  rerenderWidget5(state.widget5);
}
```

Every time user performs an action, state updates and all widgets are rerendered (expensive!). Is there any way to rerender only particular widget? I guess we can keep previous state and check for changes:

```js
function render() {
  var state = store.getState();

  if (areNotEqual(prevState.widget1, state.widget1)) {
    rerenderWidget1(state.widget1);
  }
  // ....
}
```

But I'm not sure if this is a good solution or not. If application is really big, then it would be way a lot of checks.

**Comment by gaearon** [comment]
We don't actually advise you to use Redux without a performant view layer like React, Deku, Angular 2, virtual-dom, etc. The point of this example is to show that Redux is not coupled to either and doesn't require build tools, but sure you'll need some abstraction. To clarify though, you don't need something like `areNotEqual`. A reference equality check is always enough because Redux assumes immutability. If something changes its reference changes.

**Comment by imkost** [comment]
Good point. Thanks!

File: examples/counter-vanilla/index.html (hunk 3 above)
Comment (ForbesLindesay) [inline examples/counter-vanilla/index.html]: You can also use brcdn.org for any packages that don't include umd builds in npm :)

Comment (mjackson) [inline examples/counter-vanilla/index.html]: :+1:

**Pull request merged**